osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -44,7 +46,16 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
             countOk = score.Statistics.GetValueOrDefault(HitResult.Ok);
             countMeh = score.Statistics.GetValueOrDefault(HitResult.Meh);
             countMiss = score.Statistics.GetValueOrDefault(HitResult.Miss);
-            effectiveMissCount = calculateEffectiveMissCount(osuAttributes);
+
+            if (usingClassicSliderAccuracy)
+                effectiveMissCount = calculateEffectiveMissCount(osuAttributes);
+            else
+            {
+                countSliderEndsDropped = osuAttributes.SliderCount - score.Statistics.GetValueOrDefault(HitResult.SliderTailHit);
+                // Note: HitResult.LargeTickMiss is not stored within the API or in score dumps. Do not use it!
+                countLargeTickMiss = score.MaximumStatistics.GetValueOrDefault(HitResult.LargeTickHit) - score.Statistics.GetValueOrDefault(HitResult.LargeTickHit);
+                effectiveMissCount = countMiss;
```

Comment:
I have a few issues here.

## First, this needs a unit test...

It is exceedingly difficult to parse and mentally prove whether any of this is correct. If that means you have to extract this into a `public` method and call it from a unit test, then so be it.

## Secondly, about `LargeTickMiss` storage...

Where is the assertion coming from that LTM is not stored in the API/score dumps coming from? Can you provide specific scores where this is not the case? You'll find that those scores will _also_ not store LTH, so I'm not sure this code is correct at all.

## Thirdly, regarding differences in stored statistics...

Here's a sampling of scores:

stable:
```json
{
  "data": {
    "mods": [
      {
        "acronym": "CL"
      }
    ],
    "statistics": {
      "ok": 25,
      "miss": 6,
      "great": 304
    },
    "maximum_statistics": {
      "great": 335,
      "legacy_combo_increase": 147
    }
  }
}
```

I believe the existing path is fine here - this will go through the estimator and extrapolate via slider count + combo.

lazer:
```json
{
  "data": {
    "statistics": {
      "ok": 228,
      "meh": 96,
      "miss": 531,
      "great": 984,
      "ignore_hit": 527,
      "ignore_miss": 469,
      "large_tick_hit": 498,
      "large_tick_miss": 89,
      "slider_tail_hit": 384
    },
    "maximum_statistics": {
      "great": 1839,
      "ignore_hit": 690,
      "large_tick_hit": 587,
      "slider_tail_hit": 690
    }
  }
}
```

This is where things get shaky for me.
- We can calculate dropped ends exactly via `slider_tail_hit`, and I would highly prefer that over the attributes.
- I don't understand why `large_tick_miss` would be recalculated here. This is why I'm requesting more info with the specific scores that led to the comment.

lazer + CL (classic sliders):
```json
{
  "data": {
    "mods": [
      {
        "acronym": "CL"
      }
    ],
    "statistics": {
      "ok": 65,
      "meh": 4,
      "miss": 4,
      "great": 587,
      "ignore_miss": 10,
      "large_bonus": 46,
      "small_bonus": 95,
      "small_tick_hit": 377,
      "small_tick_miss": 13,
      "large_tick_hit": 412,
      "large_tick_miss": 6
    },
    "maximum_statistics": {
      "great": 660,
      "large_bonus": 56,
      "small_bonus": 95,
      "large_tick_hit": 418,
      "small_tick_hit": 390
    }
  }
}
```

This one is perhaps the worst of the bunch.  
Right now this will go through the estimator, but we can do better because in this case dropped slider ends are represented as `small_tick_miss`.

You can use `score.IsLegacyScore` to tell this score apart from the stable one.

Reply:
If I'm remembering correctly, there was a notion that CL and stable scores should match 1:1 in PP so that was the intention with sending all classic scores down the estimation path.

Reply:
Okay, that's fair enough.

Reply:
![image](https://github.com/user-attachments/assets/f6c13c71-408a-4311-97c3-01178fa6ba92)
Can confirm the notion that lazer CL scores should match stable.

\s for why I switched to using `LargeTickHit`, I was operating mainly off of these messages:
(data.ppy.sh)
![IMG_5770](https://github.com/user-attachments/assets/a561ccc8-5c96-415c-bbdc-590b3b8d7b0d)
(API)
![IMG_5771](https://github.com/user-attachments/assets/e6a71e7e-2982-422b-95a1-a0041cf89dcf)
According to tsunyoku, I didn’t actually need to change it, so that’s my bad for operating off of an assumption.

I’ll get on making a unit test ASAP